### ISSUE TITLE
Stacking in bar and timeseries plots when negative values are present is incorrect

### ISSUE THREAD

**Issue by wrobell** [grafana/grafana#55348]
Tested with Grafana 9.1.2 and 9.1.5

The data is (data columns correspond to green, yellow, red colors in the graphs below)
![2022-09-18_22-25-16_3290x603](https://user-images.githubusercontent.com/105664/190929271-baf70613-2ca1-48c7-b471-86353328dfd2.png)
This is what I get with stacked bar plot (changing to stacked timeseries plot shows the same issue)
![2022-09-18_22-49-49_3018x783](https://user-images.githubusercontent.com/105664/190929340-24ee49a9-8f13-4bd4-a381-b2be08b3e555.png)
An example of the issue is for 11:00 and 12:00 - both yellow values should have exactly the same height (their values are 1100).

If I hide the green (negative values), then the yellow and red values are OK

![2022-09-18_22-24-29_3465x1702](https://user-images.githubusercontent.com/105664/190929306-061de6a5-41bd-48db-8d14-417c92d781fe.png)

**Comment by leeoniya** [comment]
cannot reproduce with with simple attempt:

<details><summary>neg-stacked-bars</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 346,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisCenteredZero": false,
            "axisColorMode": "text",
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "bars",
            "fillOpacity": 10,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "normal"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 21,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom",
          "showLegend": true
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "targets": [
        {
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_metric_values",
          "stringInput": "5,9,9,3"
        },
        {
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "B",
          "scenarioId": "csv_metric_values",
          "stringInput": "-12,-37,-8,-17"
        }
      ],
      "title": "Panel Title",
      "type": "timeseries"
    }
  ],
  "schemaVersion": 37,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "neg-stacked-bars",
  "uid": "cBON8t7Vk",
  "version": 2,
  "weekStart": ""
}
```
</details>

please provide your dashboard JSON and dataset.

**Comment by wrobell** [comment]
I believe it might be related to order or missing data of a  grouping field (or both). Will try to replicate it with some small example in coming days.

**Comment by wrobell** [comment]
I believe I am able to replicate the problem with the following steps.

1. Create PostgreSQL database.
2. Create a table and insert data
```
create table test (time timestamp, metric varchar, value float);
insert into test values
    ('2022-09-01 19:00', 'a', 100),
    ('2022-09-01 19:00', 'c', 150),
    ('2022-09-01 20:00', 'a', 200),
    ('2022-09-01 20:00', 'b', 300),
    ('2022-09-01 20:00', 'c', 150),
    ('2022-09-01 21:00', 'a', 250),
    ('2022-09-01 21:00', 'b', 350),
    ('2022-09-01 21:00', 'c', 150);
```
3. Import dashboard
<details>
<summary>...dashboard code...</summary>

```
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 20,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "postgres",
        "uid": "vYoi6xkRz"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisCenteredZero": false,
            "axisColorMode": "text",
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "bars",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "normal"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 15,
        "w": 16,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom",
          "showLegend": true
        },
        "tooltip": {
          "mode": "multi",
          "sort": "none"
        }
      },
      "targets": [
        {
          "datasource": {
            "type": "postgres",
            "uid": "vYoi6xkRz"
          },
          "format": "time_series",
          "group": [
            {
              "params": [
                "$__interval",
                "none"
              ],
              "type": "time"
            }
          ],
          "metricColumn": "metric",
          "rawQuery": false,
          "rawSql": "SELECT\n  $__timeGroupAlias(\"time\",$__interval),\n  metric AS metric,\n  sum(value) AS \"value\"\nFROM test\nWHERE\n  $__timeFilter(\"time\")\nGROUP BY 1,2\nORDER BY 1,2",
          "refId": "A",
          "select": [
            [
              {
                "params": [
                  "value"
                ],
                "type": "column"
              },
              {
                "params": [
                  "sum"
                ],
                "type": "aggregate"
              },
              {
                "params": [
                  "value"
                ],
                "type": "alias"
              }
            ]
          ],
          "table": "test",
          "timeColumn": "\"time\"",
          "timeColumnType": "timestamp",
          "where": [
            {
              "name": "$__timeFilter",
              "params": [],
              "type": "macro"
            }
          ]
        }
      ],
      "title": "Panel Title",
      "transformations": [
        {
          "id": "calculateField",
          "options": {
            "alias": "bx",
            "binary": {
              "left": "b",
              "operator": "*",
              "reducer": "sum",
              "right": "-1"
            },
            "mode": "binary",
            "reduce": {
              "reducer": "sum"
            },
            "replaceFields": false
          }
        },
        {
          "id": "organize",
          "options": {
            "excludeByName": {
              "a": false,
              "b": true,
              "bx": false,
              "c": false,
              "c * -1": false
            },
            "indexByName": {
              "Time": 0,
              "a": 3,
              "b": 2,
              "bx": 1,
              "c": 4
            },
            "renameByName": {
              "c * -1": "x"
            }
          }
        }
      ],
      "type": "timeseries"
    }
  ],
  "refresh": false,
  "schemaVersion": 37,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "2022-09-01T17:00:00.000Z",
    "to": "2022-09-02T22:00:00.000Z"
  },
  "timepicker": {},
  "timezone": "",
  "title": "neg-stacked-bars-pg",
  "uid": "fRssuxzRz",
  "version": 4,
  "weekStart": ""
}
```
</details>

4. Create data source and point the dashboard to the datasource.


Screenshot

![2022-09-21_20-14-39_653x747](https://user-images.githubusercontent.com/105664/191591018-c0697976-40b2-4d65-9182-5d3cfdaa761b.png)

Issues are present for 2nd and 3rd bar
- data for `a` group is not shown (values are 200 and 250)
- data for `c` is stuck at 50 (both should be 150)

Please note that all data has positive values, but the dashboard has a transformation to make `b` (`bx`) negative.

**Comment by leeoniya** [comment]
we made a debug dashboard using your instructions. in the future if you attach it directly there's no need for us to set up datasources to reproduce viz issues :)

| Key | Value |
|--|--|
| Panel | timeseries @ 9.3.0-pre (ae68f81058) |
| Grafana | 9.3.0-pre (ae68f81058) // Open Source |
<details><summary>Panel debug snapshot dashboard</summary>

```json
{
  "panels": [
    {
      "id": 2,
      "gridPos": {
        "h": 13,
        "w": 15,
        "x": 0,
        "y": 0
      },
      "type": "timeseries",
      "title": "Reproduced with embedded data",
      "transformations": [
        {
          "id": "calculateField",
          "options": {
            "alias": "bx",
            "binary": {
              "left": "b",
              "operator": "*",
              "reducer": "sum",
              "right": "-1"
            },
            "mode": "binary",
            "reduce": {
              "reducer": "sum"
            },
            "replaceFields": false
          }
        },
        {
          "id": "organize",
          "options": {
            "excludeByName": {
              "a": false,
              "b": true,
              "bx": false,
              "c": false,
              "c * -1": false
            },
            "indexByName": {
              "Time": 0,
              "a": 3,
              "b": 2,
              "bx": 1,
              "c": 4
            },
            "renameByName": {
              "c * -1": "x"
            }
          }
        }
      ],
      "datasource": {
        "type": "grafana",
        "uid": "grafana"
      },
      "fieldConfig": {
        "defaults": {
          "custom": {
            "drawStyle": "bars",
            "lineInterpolation": "linear",
            "barAlignment": 0,
            "lineWidth": 1,
            "fillOpacity": 0,
            "gradientMode": "none",
            "spanNulls": false,
            "showPoints": "auto",
            "pointSize": 5,
            "stacking": {
              "mode": "normal",
              "group": "A"
            },
            "axisPlacement": "auto",
            "axisLabel": "",
            "axisColorMode": "text",
            "scaleDistribution": {
              "type": "linear"
            },
            "axisCenteredZero": false,
            "hideFrom": {
              "tooltip": false,
              "viz": false,
              "legend": false
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "color": {
            "mode": "palette-classic"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "options": {
        "tooltip": {
          "mode": "multi",
          "sort": "none"
        },
        "legend": {
          "showLegend": true,
          "displayMode": "list",
          "placement": "bottom",
          "calcs": []
        }
      },
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "type": "grafana",
            "uid": "grafana"
          },
          "queryType": "snapshot",
          "snapshot": [
            {
              "schema": {
                "refId": "A",
                "meta": {
                  "type": "timeseries-wide",
                  "executedQueryString": "SELECT floor(extract(epoch from \"time\")/60)*60 AS \"time\", metric AS metric,  sum(value) AS \"value\" FROM test WHERE  \"time\" BETWEEN '2022-09-01T17:00:00Z' AND '2022-09-02T22:00:00Z' GROUP BY 1,2 ORDER BY 1,2",
                  "transformations": [
                    "calculateField",
                    "organize",
                    "calculateField",
                    "organize"
                  ]
                },
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {}
                  },
                  {
                    "name": "a",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64",
                      "nullable": true
                    },
                    "config": {}
                  },
                  {
                    "name": "b",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64",
                      "nullable": true
                    },
                    "config": {}
                  },
                  {
                    "name": "c",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64",
                      "nullable": true
                    },
                    "config": {}
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1662058800000,
                    1662062400000,
                    1662066000000,
                    1662069600000,
                    1662073200000,
                    1662076800000,
                    1662080400000
                  ],
                  [
                    100,
                    200,
                    250
                  ],
                  [
                    null,
                    300,
                    350
                  ],
                  [
                    150,
                    150,
                    150
                  ]
                ]
              }
            }
          ]
        }
      ]
    },
    {
      "gridPos": {
        "h": 7,
        "w": 9,
        "x": 15,
        "y": 0
      },
      "id": 5,
      "options": {
        "content": "<table width=\"100%\">\n    <tr>\n      <th width=\"2%\">Panel</th>\n      <td >timeseries @ 9.3.0-pre (ae68f81058)</td>\n    </tr>\n    <tr>\n      <th>Queries</th>\n      <td>A[postgres]</td>\n    </tr>\n    <tr>\n      <th>Transforms (2)</th>\n      <td>calculateField, organize</td>\n  </tr>\n    <tr><th>Data</th><td> 1 frames, 4 fields, 3 rows </td></tr>\n    \n    <tr>\n      <th>Grafana</th>\n      <td>9.3.0-pre (ae68f81058) // Open Source</td>\n    </tr>\n  </table>",
        "mode": "html"
      },
      "title": "Debug info",
      "type": "text"
    },
    {
      "id": 6,
      "title": "Original Panel JSON",
      "type": "text",
      "gridPos": {
        "h": 13,
        "w": 9,
        "x": 15,
        "y": 7
      },
      "options": {
        "content": "{\n  \"id\": 2,\n  \"gridPos\": {\n    \"h\": 15,\n    \"w\": 16,\n    \"x\": 0,\n    \"y\": 0\n  },\n  \"type\": \"timeseries\",\n  \"title\": \"Panel Title\",\n  \"transformations\": [\n    {\n      \"id\": \"calculateField\",\n      \"options\": {\n        \"alias\": \"bx\",\n        \"binary\": {\n          \"left\": \"b\",\n          \"operator\": \"*\",\n          \"reducer\": \"sum\",\n          \"right\": \"-1\"\n        },\n        \"mode\": \"binary\",\n        \"reduce\": {\n          \"reducer\": \"sum\"\n        },\n        \"replaceFields\": false\n      }\n    },\n    {\n      \"id\": \"organize\",\n      \"options\": {\n        \"excludeByName\": {\n          \"a\": false,\n          \"b\": true,\n          \"bx\": false,\n          \"c\": false,\n          \"c * -1\": false\n        },\n        \"indexByName\": {\n          \"Time\": 0,\n          \"a\": 3,\n          \"b\": 2,\n          \"bx\": 1,\n          \"c\": 4\n        },\n        \"renameByName\": {\n          \"c * -1\": \"x\"\n        }\n      }\n    }\n  ],\n  \"datasource\": {\n    \"uid\": \"APE38Z4Vk\",\n    \"type\": \"postgres\"\n  },\n  \"fieldConfig\": {\n    \"defaults\": {\n      \"custom\": {\n        \"drawStyle\": \"bars\",\n        \"lineInterpolation\": \"linear\",\n        \"barAlignment\": 0,\n        \"lineWidth\": 1,\n        \"fillOpacity\": 0,\n        \"gradientMode\": \"none\",\n        \"spanNulls\": false,\n        \"showPoints\": \"auto\",\n        \"pointSize\": 5,\n        \"stacking\": {\n          \"mode\": \"normal\",\n          \"group\": \"A\"\n        },\n        \"axisPlacement\": \"auto\",\n        \"axisLabel\": \"\",\n        \"axisColorMode\": \"text\",\n        \"scaleDistribution\": {\n          \"type\": \"linear\"\n        },\n        \"axisCenteredZero\": false,\n        \"hideFrom\": {\n          \"tooltip\": false,\n          \"viz\": false,\n          \"legend\": false\n        },\n        \"thresholdsStyle\": {\n          \"mode\": \"off\"\n        }\n      },\n      \"color\": {\n        \"mode\": \"palette-classic\"\n      },\n      \"mappings\": [],\n      \"thresholds\": {\n        \"mode\": \"absolute\",\n        \"steps\": [\n          {\n            \"color\": \"green\",\n            \"value\": null\n          },\n          {\n            \"color\": \"red\",\n            \"value\": 80\n          }\n        ]\n      }\n    },\n    \"overrides\": []\n  },\n  \"options\": {\n    \"tooltip\": {\n      \"mode\": \"multi\",\n      \"sort\": \"none\"\n    },\n    \"legend\": {\n      \"showLegend\": true,\n      \"displayMode\": \"list\",\n      \"placement\": \"bottom\",\n      \"calcs\": []\n    }\n  },\n  \"targets\": [\n    {\n      \"refId\": \"A\",\n      \"datasource\": {\n        \"type\": \"postgres\",\n        \"uid\": \"APE38Z4Vk\"\n      },\n      \"format\": \"time_series\",\n      \"timeColumn\": \"\\\"time\\\"\",\n      \"metricColumn\": \"none\",\n      \"group\": [],\n      \"where\": [\n        {\n          \"type\": \"macro\",\n          \"name\": \"$__timeFilter\",\n          \"params\": []\n        }\n      ],\n      \"select\": [\n        [\n          {\n            \"type\": \"column\",\n            \"params\": [\n              \"value\"\n            ]\n          }\n        ]\n      ],\n      \"rawQuery\": true,\n      \"rawSql\": \"SELECT $__timeGroupAlias(\\\"time\\\",$__interval), metric AS metric,  sum(value) AS \\\"value\\\" FROM test WHERE  $__timeFilter(\\\"time\\\") GROUP BY 1,2 ORDER BY 1,2\",\n      \"table\": \"test\",\n      \"timeColumnType\": \"timestamp\"\n    }\n  ]\n}",
        "mode": "code",
        "code": {
          "language": "json",
          "showLineNumbers": true,
          "showMiniMap": true
        }
      }
    },
    {
      "id": 3,
      "title": "Data from panel above (after transformations)",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 15,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": true,
          "refId": "A"
        }
      ]
    },
    {
      "id": 100,
      "title": "Data (before transformations)",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 24,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": false,
          "refId": "A"
        }
      ]
    }
  ],
  "schemaVersion": 37,
  "title": "Debug: Panel Title // 2022-09-27 23:09:12",
  "tags": [
    "debug",
    "debug-timeseries"
  ],
  "time": {
    "from": "2022-09-01T17:00:00.000Z",
    "to": "2022-09-02T22:00:00.000Z"
  }
}
```
</details>

![image](https://user-images.githubusercontent.com/43234/192683625-1ac073fa-d203-4544-8440-18cf7c9f6eb1.png)

i made a separate test dashboard just with your data in CSV format plus some transforms to simulate what the Postgres datasource does:

<details><summary>test-stack.json</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 349,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisCenteredZero": false,
            "axisColorMode": "text",
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "bars",
            "fillOpacity": 30,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "normal"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": [
          {
            "matcher": {
              "id": "byName",
              "options": "b"
            },
            "properties": [
              {
                "id": "custom.transform",
                "value": "negative-Y"
              }
            ]
          }
        ]
      },
      "gridPos": {
        "h": 19,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom",
          "showLegend": true
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "targets": [
        {
          "alias": "a",
          "csvContent": "time,metric,value\n2022-09-01T19:00:00Z, a, 100\n2022-09-01T19:00:00Z, c, 150\n2022-09-01T20:00:00Z, a, 200\n2022-09-01T20:00:00Z, b, 300\n2022-09-01T20:00:00Z, c, 150\n2022-09-01T21:00:00Z, a, 250\n2022-09-01T21:00:00Z, b, 350\n2022-09-01T21:00:00Z, c, 150",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_content"
        }
      ],
      "title": "Panel Title",
      "transformations": [
        {
          "id": "prepareTimeSeries",
          "options": {
            "format": "many"
          }
        },
        {
          "id": "joinByField",
          "options": {}
        },
        {
          "id": "organize",
          "options": {
            "excludeByName": {},
            "indexByName": {},
            "renameByName": {
              "value {metric=\"a\", name=\"a\"}": "a",
              "value {metric=\"b\", name=\"a\"}": "c",
              "value {metric=\"c\", name=\"a\"}": "b"
            }
          }
        }
      ],
      "type": "timeseries"
    }
  ],
  "refresh": false,
  "schemaVersion": 37,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "2022-09-01T18:13:18.658Z",
    "to": "2022-09-01T22:07:45.586Z"
  },
  "timepicker": {},
  "timezone": "",
  "title": "test-stack",
  "uid": "piJUyWV4z",
  "version": 4,
  "weekStart": ""
}
```
</details>

long story short, i think all you need to do for your case is to remove all Transformations, and add a field override to be for `Negative Y`, as below. we'll continue to investigate why your specific set of options doesnt result in the same thing.

![image](https://user-images.githubusercontent.com/43234/192685149-2c3cde28-215f-49e1-8ccf-a15522970414.png)

**Comment by wrobell** [comment]
Thanks for the tip. This seems to work for Timeseries panel, but graph style override seems to be not available for Bar Chart panel?

**Comment by leeoniya** [comment]
> not available for Bar Chart panel?

should be fixed by https://github.com/grafana/grafana/pull/55930

**Comment by wrobell** [comment]
@leeoniya Thanks for the workaround once again.

But I believe the ticket should not be closed. There is still something wrong with the visualization when transformations are involved.

**Comment by marko-savic** [comment]
The problem still exists in 9.3.
Render is broken when it happens that one series is negative and one positive at some point in the view.
![image](https://user-images.githubusercontent.com/6500658/210005248-b2af6ba6-f4ec-4eda-a82c-fd612074ef2c.png)

![image](https://user-images.githubusercontent.com/6500658/210004978-41074852-0899-45b3-a58f-aeafcaa3981c.png)

**Comment by maufc** [comment]
Same here. 
And making an override is not an option when values can go positive and negative.

**Comment by zoltanbedi** [comment]
@maufc can you please open a new issue with steps or dashboard to reproduce this?

**Comment by leeoniya** [comment]
i think the issue you're looking for is https://github.com/grafana/grafana/issues/51011

for bar stacking, i think a fix is feasible since there is no zero-crossover point, but not for lines/areas. (see discussion in linked issue)
